modules/userId/index.js

```diff
@@ -590,16 +590,17 @@ function refreshUserIds(options, callback) {
   initializeSubmodulesAndExecuteCallbacks(function() {
     let consentData = gdprDataHandler.getConsentData()
 
-    const storedConsentData = getStoredConsentData();
-    setStoredConsentData(consentData);
-
     // gdpr consent with purpose one is required, otherwise exit immediately
     let {userIdModules, hasValidated} = validateGdprEnforcement(submodules, consentData);
     if (!hasValidated && !hasGDPRConsent(consentData)) {
```

Comment:
Regarding https://github.com/prebid/Prebid.js/issues/6386#issuecomment-1402272770

If gdpr applies and there is no consent, here you get `{userIdModules: [], hasValidated: true}`, in other words, this check passes and we go ahead to write the cookie on line 255 - even though none of the userId modules will be initialized as they all failed the consent check.